test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -179,25 +179,27 @@ func (s *snapshottableTestSuite) DefineTests(driver storageframework.TestDriver,
 			// In the following code by checking if the PV is not in the node.Status.VolumesInUse field we
 			// ensure that the volume is not used by the node anymore (an indicator that NodeUnstageVolume has
 			// already finished)
-			if framework.NodeOSDistroIs("windows") {
-				nodeName := pod.Spec.NodeName
-				gomega.Expect(nodeName).NotTo(gomega.BeEmpty(), "pod.Spec.NodeName must not be empty")
-
-				ginkgo.By(fmt.Sprintf("[init] waiting until the node=%s is not using the volume=%s", nodeName, pv.Name))
-				success := storageutils.WaitUntil(framework.Poll, f.Timeouts.PVDelete, func() bool {
-					node, err := cs.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
-					framework.ExpectNoError(err)
-					volumesInUse := node.Status.VolumesInUse
-					framework.Logf("current volumes in use: %+v", volumesInUse)
-					for i := 0; i < len(volumesInUse); i++ {
-						if strings.HasSuffix(string(volumesInUse[i]), pv.Name) {
-							return false
-						}
+			//
+			// We need to force this for non-Windows distros as well, as NodeUnpublishVolume only removes
+			// the bind mount, which doesn't force the caches to flush. It's possible to create empty
+			// snapshots if we don't wait (see #101279 for details).
```

Comment:
Can you please incorporate this comment into the comment above?

```
			//   - NodeUnpublishVolume: unbinds the bind mount from the container
			//     - Linux: the bind mount is removed, which does not flush any cache
...
			//   - NodeUnstageVolume: unmount the global mount
			//     - Linux: disk is unmounted and all caches flushed.
```

Reply:
These changes have been made.